### PR TITLE
Allow dependencies to be defined on params

### PR DESCRIPTION
**Pull request by philipgough**
Relates to https://github.com/openshift/ansible-service-broker/issues/859

### CODE REVIEW

**Comment by dymurray** [comment]
I have read through the attached issue and I like the solution that you guys came up with. Since this will involve an addition to the Bundle spec this type of change will require a version bump of the Bundle Spec along with documentation updates (see [here](https://github.com/ansibleplaybookbundle/ansible-playbook-bundle/blob/master/docs/developers.md#service-bundle-spec-file) for current documentation around Bundle spec).

**Comment by dymurray** [comment]
From discussing at the community meeting we agreed that this will require a minor version bump of the Bundle spec bringing us to version `1.1` (per [our docs](https://github.com/ansibleplaybookbundle/ansible-playbook-bundle/blob/master/docs/developers.md#service-bundle-spec-versioning) since this is the addition of an optional field). Please update `developers.md` with an example of using this and defining its use in the Bundle spec as I linked above.

File: bundle/types.go [hunk 1]
```diff
@@ -66,11 +66,17 @@ type ParameterDescriptor struct {
...
-	Required     bool     `json:"required"`
-	Updatable    bool     `json:"updatable"`
-	DisplayType  string   `json:"displayType,omitempty" yaml:"display_type,omitempty"`
-	DisplayGroup string   `json:"displayGroup,omitempty" yaml:"display_group,omitempty"`
+	Enum         []string     `json:"enum,omitempty"`
+	Required     bool         `json:"required"`
+	Updatable    bool         `json:"updatable"`
+	DisplayType  string       `json:"displayType,omitempty" yaml:"display_type,omitempty"`
+	DisplayGroup string       `json:"displayGroup,omitempty" yaml:"display_group,omitempty"`
+	Dependencies []Dependency `json:"dependencies,omitempty" yaml:"dependencies,omitempty"`
```
Comment (philipgough) [inline bundle/types.go:74]: We can only handle a single dependency here currently in the broker due to limitations in the frontend lib but I think to future proof where we may be able to do this if we can extend that lib, leaving this as a slice makes sense IMO. What do people think?

The spec itself should be able to support multiple dependencies if my understanding of it is correct.

cc @david-martin

**Comment by dymurray** [comment]
Thanks for doing a doc PR!

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,6 @@
 vendor*
 coverage-all.out
 coverage.out
+
+# IDE baggage
+.idea/
```
Comment (jmrodri) [inline .gitignore:6]: this is acceptable

File: bundle/types.go (hunk 1 above)
Comment (jmrodri) [reply bundle/types.go:74]: Agreed @PhilipGough  allow multiple dependencies.

**Review by jmrodri — APPROVED** [review]
I'm actually okay with this, I don't see any issues.

**Review by dymurray — APPROVED** [review]

**Pull request merged**